google-cloud-firestore/lib/google/cloud/firestore/database.rb

```diff
@@ -0,0 +1,1013 @@
+        #
+        #   # Update a document
+        #   firestore.update("cities/NYC", { name: "New York City" },
+        #                    update_time: last_updated_at)
+        #
+        def update doc, data, update_time: nil
+          batch { |b| b.update doc, data, update_time: update_time }
+        end
+
+        ##
+        # Deletes a document from the database.
+        #
+        # @param [String, Document::Reference] doc A string representing the
+        #   path of the document, or a document reference object.
+        # @param [Boolean] exists Whether the document must exist. When `true,
+        #   the document must exist or an error is raised. Default is `false`.
+        #   Optional.
+        # @param [Time] update_time When set, the document must have been last
+        #   updated at that time. Optional.
+        #
+        # @return [Time] The time the changes were committed
+        #
+        # @example Delete a document using a document path:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Delete a document
+        #   firestore.delete "cities/NYC"
+        #
+        # @example Delete a document using a document reference:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   # Delete a document
+        #   firestore.delete nyc_ref
+        #
+        # @example Delete a document using `exists`:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Delete a document
+        #   firestore.delete "cities/NYC", exists: true
+        #
+        # @example Delete a document using `update_time`:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   last_updated_at = Time.now - 42 # 42 seconds ago
+        #
+        #   # Delete a document
+        #   firestore.delete "cities/NYC", update_time: last_updated_at
+        #
+        def delete doc, exists: nil, update_time: nil
+          batch { |b| b.delete doc, exists: exists, update_time: update_time }
+        end
+
+        ##
+        # Create a transaction to perform multiple reads and changes that are
+        # executed atomically at a single logical point in time in a database.
+        #
+        # All changes are accumulated in memory until the block completes.
+        # Transactions will be automatically retried when possible. See
+        # {Transaction}.
+        #
+        # @see https://firebase.google.com/docs/firestore/manage-data/transactions
+        #   Transactions and Batched Writes
+        #
+        # @param [String] previous_transaction The transaction identifier of a
```

Comment:
The transaction retry is an implementation detail that should not be part of the public API.